src/app/components/ElectronicDelivery/ElectronicDeliveryForm.jsx

```diff
@@ -19,13 +19,17 @@ class ElectronicDeliveryForm extends React.Component {
     this.state = {
       form: !_isEmpty(this.props.form) ? this.props.form :
         {
-          name: '',
-          email: '',
+          emailAddress: '',
```

Comment:
I guess it is because the fields' names. The names before didn't match what the API wants.